oai_agents/agents/base_agent.py

```diff
@@ -401,30 +401,23 @@ def evaluate(self, eval_agent, num_eps_per_layout_per_tm=5, visualize=False, tim
         rew_per_layout_per_teamtype = {}
         rew_per_layout = {}
 
-        '''
-        dict 
-        teammates_collection = {
-            'layout_name': {
-                'TeamType.HIGH_FIRST': [[agent1, agent2], ...],
-                'TeamType.MEDIUM_FIRST': [[agent3, agent4], ...],
-                'TeamType.LOW_FIRST': [[agent5, agent6], ...],
-                'TeamType.RANDOM': [[agent7, agent8], ...],
-            },
-        }
-        '''
+        # To reduce evaluation time: instead of evaluating all players, we randomly select either two or three of player positions for evaluation
+        # This is outside of the for loop, meaning that each time we evaluate the same player positions across all layouts for a fair comparison
+        selected_p_indexes = random.sample(range(self.args.num_players), min(random.choice([2, 3]), self.args.num_players))
```

Comment:
Is there a motivation or a clear benefit to evaluate two or three positions instead of always three positions when there are more than three players in games?

Reply:
Yes, I wanted to reduce the training time for 3 player layouts as well because they also take a lot of time (thus either 2 or 3). Moreover, in case of 5 player layouts I wanted to reduce the training time even further but always evaluating with 2 agents is too limited, whereas only evaluating with 3 agents still take a lot of time that's why I randomly select either 2 or 3.

Reply:
I see. It makes sense to me. Thank you. 

If we want both (1) evaluation to provide enough information (2) not take a long time on evaluation, we have the other choice: We can reduce the evaluation frequency, and still evaluate our agents with 3 or even more different positions.

Reply:
Like you said, if we only evaluate with 2 agents, the evaluation is limited. This also raise a concern on the "best" agent we picked. The "best" agent we picked will be based on part of evaluation instead of full evaluation.

Reply:
changed this to:
1) only select 3 agents for evaluation
2) evaluate agents every 15 epochs instead of every 5 epochs. 
In practice it will not be evaluated every 15 epochs but rather when a better model is found or it should save a checkpoint: [link](https://github.com/HIRO-group/multiHRI/blob/8788c91a2e6402263e784fb8fc869000357c5cbb/oai_agents/agents/rl.py#L270)

what do you think?